### ISSUE TITLE
Cannot load ruleset ruleset.xml: 2 XML validation errors occurred

### ISSUE THREAD

**Issue by vivelan** [pmd/pmd#4892]
Hi, 

I've encountered the error below starting today; however, until yesterday, I hadn't experienced this issue. I've provided my ruleset and pipeline code. Could you please assist me in resolving this error?

Pipeline : 

```yml
# Unique name for this workflow
name: PMD Code Scan

# Definition when the workflow should run
on:
    # The workflow will run whenever an event happens on a pull request
    pull_request:
      # The events are that a PR is opened, or when a commit is pushed
      # to a branch that has an existing pull request
      types: [opened, synchronize]
      # The branches filter allows to specify that this workflow should only
      # run if the branch name is as listed below. This way we prevent this workflow
      # from running when PRs are opened on other branches
      branches: [ environment/*/dev , environment/qa ]
      # We only care about changes to the force-app directory, which is the
      # root directory of the sfdx project. This prevents the job from running
      # when changing non-salesforce files (like this yml file).
      paths:
        - 'force-app/**'
            

# Jobs to be executed when the above conditions are met
jobs:
    # This is the name of the job. You can give it whatever name you want
    Run-PMD-Code-Scan:
        # As mentioned in the blog post, this job runs inside a VM. Here we
        # can specify which OS this VM should run on. 
        # In this case, we are going to run our commands on the latest version
        # of ubuntu
        runs-on: ubuntu-latest
        permissions:
                    contents: read
                    pull-requests: write
                    # required for all workflows
                    security-events: write
                    # only required for workflows in private repositories
                    actions: read
                    
        if: ${{ github.actor != 'dependabot[bot]' }}
        steps:
      
            # Now we install nodejs in the VM, and specify version 14
            - uses: actions/setup-node@v4
              with:
                node-version: '18'

            # The idea is that the VM can access your remote repository
            # because your repository is an sfdx project.
            # This is a default action that allows us to enter the root 
            # directory of the repository

            # Make sure to specify fetch-depth:0. This allows us to
            # access previous commits that have been pushed to the repository.

            # We'll need this later when we try to figure out which metadata has 
            # changed between commits, so that we can only deploy that metadata
            # to the destination org

            - name: 'Checkout source code'
              uses: actions/checkout@v4
              with:
                fetch-depth: 0
            
            # Run PMD Scan
            - name: 'Run PMD Scan'
              uses: pmd/pmd-github-action@main
              id: pmd
              with: 
                    sourcePath: 'force-app'
                    rulesets: 'ruleset.xml'    
                    analyzeModifiedFilesOnly: true
                    createGitHubAnnotations: true 
                    
            - name: Fail build if there are violations
              if: steps.pmd.outputs.violations != 0
              run: exit 1              
```

RuleSet:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<ruleset xmlns="http://pmd.sourceforge.net/ruleset/2.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" name="Default ruleset used by the CodeClimate Engine for Salesforce.com Apex" xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 https://pmd.sourceforge.io/ruleset_2_0_0.xsd">
   <description> Default ruleset for code scanning.</description>
   
   <rule ref="category/apex/bestpractices.xml/ApexAssertionsShouldIncludeMessage" />
   <rule ref="category/apex/bestpractices.xml/ApexUnitTestClassShouldHaveAsserts" />
   <rule ref="category/apex/design.xml/AvoidDeeplyNestedIfStmts" >
      <properties>
        <property name="problemDepth" value="7" />
      </properties>
   </rule>
   <rule ref="category/apex/bestpractices.xml/ApexUnitTestShouldNotUseSeeAllDataTrue" />
   <rule ref="category/apex/bestpractices.xml/AvoidLogicInTrigger" />
   <rule ref="category/apex/bestpractices.xml/DebugsShouldUseLoggingLevel" />
   <rule ref="category/apex/bestpractices.xml/UnusedLocalVariable" />
   <rule ref="category/apex/design.xml/UnusedMethod" />
   <rule ref="category/apex/errorprone.xml/AvoidHardcodingId" />
   <rule ref="category/apex/errorprone.xml/EmptyCatchBlock" />
   <rule ref="category/apex/performance.xml/OperationWithLimitsInLoop" />
   <rule ref="category/apex/performance.xml/AvoidDmlStatementsInLoops" />
   <rule ref="category/apex/performance.xml/AvoidSoqlInLoops" />
   <rule ref="category/apex/security.xml/ApexDangerousMethods" />
   <rule ref="category/apex/security.xml/ApexInsecureEndpoint" />
   <rule ref="category/apex/security.xml/ApexOpenRedirect" />
   <rule ref="category/apex/security.xml/ApexSOQLInjection" />
   <rule ref="category/apex/security.xml/ApexSuggestUsingNamedCred" />
   <rule ref="category/apex/security.xml/ApexXSSFromEscapeFalse" />
   <rule ref="category/apex/security.xml/ApexXSSFromURLParam" />
   <rule ref="category/apex/security.xml/ApexSharingViolations" />   
</ruleset>
```

Error:

```
Error:  Error at ruleset.xml:20:4
 [18](https://github.com/shs-it/CRMPortale/actions/runs/8431583506/job/23089241725?pr=2401#step:4:19)|    <rule ref="category/apex/errorprone.xml/EmptyCatchBlock" />
 [19](https://github.com/shs-it/CRMPortale/actions/runs/8431583506/job/23089241725?pr=2401#step:4:20)|    <rule ref="category/apex/performance.xml/OperationWithLimitsInLoop" />
 [20](https://github.com/shs-it/CRMPortale/actions/runs/8431583506/job/23089241725?pr=2401#step:4:21)|    <rule ref="category/apex/performance.xml/AvoidDmlStatementsInLoops" />
        ^^^^^ Unable to find referenced rule AvoidDmlStatementsInLoops; perhaps the rule name is misspelled?

 [21](https://github.com/shs-it/CRMPortale/actions/runs/8431583506/job/23089241725?pr=2401#step:4:22)|    <rule ref="category/apex/performance.xml/AvoidSoqlInLoops" />
 [22](https://github.com/shs-it/CRMPortale/actions/runs/8431583506/job/23089241725?pr=2401#step:4:23)|    <rule ref="category/apex/security.xml/ApexDangerousMethods" />
Error:  Error at ruleset.xml:21:4
 19|    <rule ref="category/apex/performance.xml/OperationWithLimitsInLoop" />
 20|    <rule ref="category/apex/performance.xml/AvoidDmlStatementsInLoops" />
 21|    <rule ref="category/apex/performance.xml/AvoidSoqlInLoops" />
        ^^^^^ Unable to find referenced rule AvoidSoqlInLoops; perhaps the rule name is misspelled?

 22|    <rule ref="category/apex/security.xml/ApexDangerousMethods" />
 [23](https://github.com/shs-it/CRMPortale/actions/runs/8431583506/job/23089241725?pr=2401#step:4:24)|    <rule ref="category/apex/security.xml/ApexInsecureEndpoint" />
Error:  Cannot load ruleset ruleset.xml: 2 XML validation errors occurred
PMD exited with 1
Error: Unexpected end of JSON input
```

Thanks

**Comment by vivelan** [comment]
RuleSet:

<?xml version="1.0" encoding="UTF-8"?>
<ruleset xmlns="http://pmd.sourceforge.net/ruleset/2.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" name="Default ruleset used by the CodeClimate Engine for Salesforce.com Apex" xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 https://pmd.sourceforge.io/ruleset_2_0_0.xsd">
   <description> Default ruleset for code scanning.</description>
   
   <rule ref="category/apex/bestpractices.xml/ApexAssertionsShouldIncludeMessage" />
   <rule ref="category/apex/bestpractices.xml/ApexUnitTestClassShouldHaveAsserts" />
   <rule ref="category/apex/design.xml/AvoidDeeplyNestedIfStmts" >
      <properties>
        <property name="problemDepth" value="7" />
      </properties>
   </rule>
   <rule ref="category/apex/bestpractices.xml/ApexUnitTestShouldNotUseSeeAllDataTrue" />
   <rule ref="category/apex/bestpractices.xml/AvoidLogicInTrigger" />
   <rule ref="category/apex/bestpractices.xml/DebugsShouldUseLoggingLevel" />
   <rule ref="category/apex/bestpractices.xml/UnusedLocalVariable" />
   <rule ref="category/apex/design.xml/UnusedMethod" />
   <rule ref="category/apex/errorprone.xml/AvoidHardcodingId" />
   <rule ref="category/apex/errorprone.xml/EmptyCatchBlock" />
   <rule ref="category/apex/performance.xml/OperationWithLimitsInLoop" />
   <rule ref="category/apex/performance.xml/AvoidDmlStatementsInLoops" />
   <rule ref="category/apex/performance.xml/AvoidSoqlInLoops" />
   <rule ref="category/apex/security.xml/ApexDangerousMethods" />
   <rule ref="category/apex/security.xml/ApexInsecureEndpoint" />
   <rule ref="category/apex/security.xml/ApexOpenRedirect" />
   <rule ref="category/apex/security.xml/ApexSOQLInjection" />
   <rule ref="category/apex/security.xml/ApexSuggestUsingNamedCred" />
   <rule ref="category/apex/security.xml/ApexXSSFromEscapeFalse" />
   <rule ref="category/apex/security.xml/ApexXSSFromURLParam" />
   <rule ref="category/apex/security.xml/ApexSharingViolations" />   
</ruleset>

**Comment by jsotuyod** [comment]
You are [not specifying a `version` for PMD](https://github.com/pmd/pmd-github-action/?tab=readme-ov-file#inputs), the action defaults to using the latest one available, and since last Friday, that's PMD 7.0.0

PMD 7.0.0 revamped several rules. The details are in the changelog but [for reference](https://docs.pmd-code.org/latest/pmd_release_notes_pmd7.html#removed-rules):

The rules `performance.xml/AvoidSoqlInLoops`, `performance.xml/AvoidSoslInLoops` and `performance.xml/AvoidDmlStatementsInLoops` have been merged into [`performance.xml/OperationWithLimitsInLoop`](https://docs.pmd-code.org/latest/pmd_rules_apex_performance.html#operationwithlimitsinloop)

You either update the ruleset to accommodate these changes, or you set the version on your pipeline to stay at the no longer supported PMD 6.55.0.
